src/npc.cpp

```diff
@@ -327,8 +348,10 @@ void Npc::onThink(uint32_t interval)
 		m_npcEventHandler->onThink();
 	}
 
-	if (getTimeSinceLastMove() >= walkTicks) {
-		addEventWalk();
+	if (!isIdle) {
+		if (getTimeSinceLastMove() >= walkTicks) {
```

Comment:
No, not really. I just surrounded the previous code with an if. I might as well concatenate the condition checks.